timesketch/api/v1/resources.py

```diff
@@ -338,12 +373,66 @@ def post(self, sketch_id):
         form = SaveViewForm.build(request)
         if form.validate_on_submit():
             sketch = Sketch.query.get_with_acl(sketch_id)
+            new_searchtemplate = form.new_searchtemplate.data
```

Comment:
Can you factor this out into a a separate method? The logic here is hard to follow with everything in a conditional like this.